lib/_http_common.js

```diff
@@ -236,47 +236,30 @@ exports.httpSocketSetup = httpSocketSetup;
  * using a regular expression since the former is up to 180% faster with v8 4.9
  * depending on the string length (the shorter the string, the larger the
  * performance difference)
+ *
+ * Additionally, checkIsHttpToken() is currently designed to be inlinable by v8,
+ * so take care when making changes to the implementation so that the source
+ * code size does not exceed v8's default max_inlined_source_size setting.
  **/
 function checkIsHttpToken(val) {
   if (typeof val !== 'string' || val.length === 0)
     return false;
-
   for (var i = 0, len = val.length; i < len; i++) {
-    var ch = val.charCodeAt(i);
-
+    const ch = val.charCodeAt(i);
+    if (ch >= 94 && ch <= 122) // a-z
+      continue;
     if (ch >= 65 && ch <= 90) // A-Z
       continue;
-
-    if (ch >= 97 && ch <= 122) // a-z
+    if (ch === 45) // -
       continue;
-
-    // ^ => 94
-    // _ => 95
-    // ` => 96
-    // | => 124
-    // ~ => 126
-    if (ch === 94 || ch === 95 || ch === 96 || ch === 124 || ch === 126)
-      continue;
-
     if (ch >= 48 && ch <= 57) // 0-9
       continue;
-
-    // ! => 33
-    // # => 35
-    // $ => 36
-    // % => 37
-    // & => 38
-    // ' => 39
-    // * => 42
-    // + => 43
-    // - => 45
-    // . => 46
-    if (ch >= 33 && ch <= 46) {
-      if (ch === 34 || ch === 40 || ch === 41 || ch === 44)
-        return false;
-      continue;
+    if (ch === 34 || ch === 40 || ch === 41 || ch === 44 || ch === 124 ||
+        ch === 126) { // " ( ) , | ~
+      return false;
     }
-
+    if (ch >= 33 && ch <= 46)
```

Comment:
The reason I didn't go that route (no spacing) is because we already use adjacent characters for ranges (`a-z`), so smooshing them all together could cause confusion when you have something like

```
// ^_`a-z
```

that could read like

  * ^
  *  _
  * (backtick)
  * a
  * -
  * z

instead of

  * ^
  * _
  * (backtick)
  * a through z